hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/SCMNodeManager.java

```diff
@@ -151,7 +152,9 @@ private void unregisterMXBean() {
    */
   @Override
   public List<DatanodeDetails> getNodes(NodeState nodestate) {
-    return nodeStateManager.getNodes(nodestate).stream()
+    return nodeStateManager.getNodes(
+        new NodeStatus(HddsProtos.NodeOperationalState.IN_SERVICE, nodestate))
+        .stream()
         .map(node -> (DatanodeDetails)node).collect(Collectors.toList());
   }
```

Comment:
In the final patch, should change the Node Query Function? so we can say, get me all the nodes that are in service and healthy, or all nodes in maintenance mode but dead? Let us Add that feature when we need it. I am ok with all operations mapping to IN_SERVICE for now.

Reply:
Yea I need to fix the query function. I can imagine we will need things like all IN_MAINT nodes (ignoring healthy, dead etc) or all dead (ignore op state). Right now that is not possible to query until I figure out how to enhance the interface.